From 25c1774d580fbd8853bdc0a1905cec48b0b3ab15 Mon Sep 17 00:00:00 2001
From: Ingmar Vanhassel <ingmar@exherbo.org>
Date: Tue, 18 Nov 2008 02:02:19 +0100
Subject: [PATCH] Fix compilation when -fno-exceptions is used.

Source: https://svn.boost.org/trac/boost/ticket/2499
Upstream: Pending
Reason: Fix compilation when -fno-exceptions is used, breaks openoffice
compilation.
---
 boost/function/function_template.hpp |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/boost/function/function_template.hpp b/boost/function/function_template.hpp
index 584abe9..36b619b 100644
--- a/boost/function/function_template.hpp
+++ b/boost/function/function_template.hpp
@@ -950,10 +950,10 @@ namespace boost {
           f.vtable->manager(f.functor, this->functor,
                             boost::detail::function::move_functor_tag);
 		  f.vtable = 0;
-#if !defined(BOOST_NO_EXCEPTIONS)      
         } else {
           clear();
         }
+#if !defined(BOOST_NO_EXCEPTIONS)      
       } catch (...) {
         vtable = 0;
         throw;
-- 
1.6.0.4

